### ISSUE TITLE
`lib.hm` missing when using plasma-manager flake

### ISSUE THREAD

**Issue by IFcoltransG** [nix-community/plasma-manager#128]
I've solved my own issue, but I want to create an issue so others who are equally new to NixOS can find the resolution. I have a small flakes setup with `home-manager` and was trying to add `programs.plasma.enable = true;` to my `home.nix` module.

My first attempts hit this error:
```
error: attribute 'hm' missing
       at /nix/store/...-source/modules/apps/kate/default.nix:181:93:
          180|     # In case of using a custom theme, check that there is no name collision
          181|     home.activation.checkKateTheme = lib.mkIf (cfg.enable && cfg.editor.theme.src != null) (lib.hm.dag.entryBefore [ "writeBoundary" ]
             |                                                                                             ^
          182|       # No `$DRY_RUN_CMD`, since even a dryrun should fail if checks fail
```
It's looking for a `hm` in `lib`. It seems that `lib` is a built-in part of Nix, but `home-manager` shadows it with an extended `lib` when you're declaring options inside `home-manager.users.<user>`. I believe the error came from trying to declare `plasma-manager` as a top-level module in my flake, but I've forgotten the exact flake. It needs to be imported within my `home.nix` module.

I had some confusion about whether I was importing `plasma-manager` (the flake) or `plasma-manager.homeManagerModules.plasma-manager` (the module) but eventually I discovered that I needed to have the following in my `home.nix`...
```nix
imports = [
  plasma-manager.homeManagerModules.plasma-manager
];
```
...as well as `plasma-manager` inside my module parameters (`{ plasma-manager, ... }:` at the start of the file) so that it would be passed into `home.nix` from the `inputs` that I had given it in my `home-manager.extraSpecialArgs`. For this, I also had to make sure that the `plasma-manager` flake I had declared in my `flake.nix` `inputs` was passed to the `outputs` module as an argument (so my parameters looked like `outputs = { self, nixpkgs, plasma-manager, ...}:`). However, my first attempt led to an infinite recursion error.
```
error: infinite recursion encountered
       at /nix/store/...-source/lib/modules.nix:508:28:
          507|         builtins.addErrorContext (context name)
          508|           (args.${name} or config._module.args.${name})
             |                            ^
          509|       ) (lib.functionArgs f);
```

I eventually solved it by consulting [this Discourse thread](https://discourse.nixos.org/t/how-import-module-from-github-into-home-manager-config/28533).

I'm not sure exactly where the infinite recursion comes from, but I resolved it by explicitly re-binding `plasma-manager` in the arguments passed to `home-manager`.
```nix
extraSpecialArgs = {
  inherit inputs;
  plasma-manager = inputs.plasma-manager;
};
```
I presume that causes `plasma-manager.homeManagerModules.plasma-manager` to be defined outside of `home-manager` and then passed into it, rather than forming part of the definition of `home manager` itself. If it's part of the definition of `home-manager`, then that would explain not being able to access `.homeManagerModules` without getting infinite recursion.

**Comment by IFcoltransG** [comment]
## TL;DR solution:

`flakes.nix`:
```nix
{
  ...
  inputs = {
    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";

    home-manager = {
      url = "github:nix-community/home-manager";
      inputs.nixpkgs.follows = "nixpkgs";
    };

    plasma-manager = {
      url = "github:pjones/plasma-manager";
      inputs.nixpkgs.follows = "nixpkgs";
      inputs.home-manager.follows = "home-manager";
    };
  };

  outputs =
  { self, nixpkgs, plasma-manager, ... }@inputs: {
    nixosConfigurations.default = nixpkgs.lib.nixosSystem {
      specialArgs = {inherit inputs;};
      modules = [
        ./path/to/configuration.nix
        inputs.home-manager.nixosModules.default
      ];
    };
  };
}
```

`configuration.nix`:
```nix
{ inputs, ... }: {
  ...
  home-manager = {
    extraSpecialArgs = {
      inherit inputs;
      plasma-manager = inputs.plasma-manager;
    };
    users = {
      myUserName = import ./path/to/home.nix;
    };
  };
  ...
}
```

`home.nix`:
```nix
{ plasma-manager, ... }:

{
  imports = [
    plasma-manager.homeManagerModules.plasma-manager
  ];
  ...
  programs.plasma.enable = true;
  ...
}
```
